server/py/services/api/crud/model_monitoring/deployment.py

```diff
@@ -1347,60 +1360,43 @@ async def create_model_endpoints(self, function: dict, function_name: str):
                 mm_constants.EventFieldType.SAMPLING_PERCENTAGE, 100
             ),
         )  # model endpoint, creation strategy, model path
-        router_model_endpoints_instructions: list[
+        semaphore = Semaphore(50)  # Limit concurrent tasks
+        tasks = []
+        batchsize = 500
+        for i in range(0, len(model_endpoints_instructions), batchsize):
+            batch = model_endpoints_instructions[i : i + batchsize]
+            tasks.append(
+                MonitoringDeployment._create_model_endpoint_limited(
+                    semaphore, batch, project
+                )
+            )
```

Comment:
These are actually not tasks. They're coroutines. A task can be created from a coroutine as follows (not necessary here because `asyncio.gather()` does it for us):
```python
task = asyncio.create_task(coroutine)
```
https://docs.python.org/3/library/asyncio-task.html#creating-tasks
```suggestion
            coroutines.append(
                MonitoringDeployment._create_model_endpoint_limited(
                    semaphore, batch, project
                )
            )
```